src/prefect_server/api/flows.py

```diff
@@ -194,11 +205,39 @@ async def create_flow(
                 "heartbeat_enabled": True,
                 "lazarus_enabled": True,
                 "version_locking_enabled": False,
+                "idempotency_key": idempotency_key,
             },
         ).insert()
     else:
         flow_group_id = flow_group.id
 
+        # check idempotency key and  early if we find a matching key and flow,
```

Comment:
```suggestion
        # check idempotency key and exit early if we find a matching key and flow,
```

make sense this way?

Reply:
Yep! 🙂